systemd.services scripts: Allow languages besides shell

zopa opened the pull request:
This lets me write

```
systemd.services.hstest.script = {
  text = ''
    {-# LANGUAGE OverloadedStrings #-}
    import Prelude hiding (FilePath)
    import Turtle

    main = do
      output "/root/turtleout" (fromString . show <$> ls "/var")
      sleep 600
  '';
language = "haskell";
env = pkgs.haskellPackages.ghcWithPackages (p: [ p.turtle ]);
```

  };

Supports only haskell at the moment, but it won't be too hard to add others.

benley commented on nixos/modules/system/boot/systemd-unit-options.nix:
```diff
@@ -209,10 +237,16 @@ in rec {
+        type = with types; either lines (submodule subscript);
+        default = "";
+        description = ''
+          Code executed as the service's main process.
```
Would it be unreasonable to simply include a shebang line at the top of the script segment, rather than adding a bunch of plumbing to explicitly support each language?

e.g. would `#!${haskellPackages.ghc}/bin/runghc` be good enough?

benley commented on pkgs/games/wesnoth/default.nix:
```diff
@@ -1,21 +1,21 @@
 { stdenv, fetchurl, cmake, SDL, SDL_image, SDL_mixer, SDL_net, SDL_ttf, pango
 , gettext, zlib, boost, freetype, libpng, pkgconfig, lua, dbus, fontconfig, libtool
-, fribidi, asciidoc }:
+, fribidi, asciidoc, libvorbis }:
```
it looks like this file wasn't meant to be included in this PR

zopa committed:
google-cloud-sdk: Add various modules to PYTHONPATH

This fixes a sequence of missing-module errors seen when calling
`gcloud auth authenticate-service-acount`.
